activemodel/lib/active_model/attribute_assignment.rb

```diff
@@ -30,11 +37,92 @@ def assign_attributes(new_attributes)
       return if new_attributes.blank?
 
       attributes = new_attributes.stringify_keys
+
+      multi_parameter_attributes = {}
+
+      attributes.each do |k, v|
+        if k.include?("(")
+          multi_parameter_attributes[k] = attributes.delete(k)
+        end
+      end
       _assign_attributes(sanitize_for_mass_assignment(attributes))
+
+      assign_multiparameter_attributes(multi_parameter_attributes) unless multi_parameter_attributes.empty?
+    end
+
+    module ClassMethods
+      mattr_accessor :typecasted
+
+      def typecast_attribute(attribute, klass)
+        self.typecasted ||= {}
+        typecasted[attribute] = klass
+      end
     end
 
     private
 
+    # Instantiates objects for all attribute classes that needs more than one constructor parameter. This is done
+    # by calling new on the column type or aggregation type (through composed_of) object with these parameters.
+    # So having the pairs written_on(1) = "2004", written_on(2) = "6", written_on(3) = "24", will instantiate
+    # written_on (a date type) with Date.new("2004", "6", "24"). You can also specify a typecast character in the
+    # parentheses to have the parameters typecasted before they're used in the constructor. Use i for Fixnum and
+    # f for Float. If all the values for a given attribute are empty, the attribute will be set to +nil+.
+    def assign_multiparameter_attributes(pairs)
+      execute_callstack_for_multiparameter_attributes(
+        extract_callstack_for_multiparameter_attributes(pairs)
+      )
+    end
+
+    def execute_callstack_for_multiparameter_attributes(callstack)
+      errors = []
+      callstack.each do |name, values_with_empty_parameters|
+        begin
+          if values_with_empty_parameters.each_value.all?(&:nil?)
```

Comment:
you can do `values_with_empty_parameters.values.none?`

`[nil, nil].none?` => true